geth/jail/console/console.go

```diff
@@ -0,0 +1,130 @@
+package console
+
+import (
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/robertkrimen/otto"
+	"github.com/status-im/status-go/geth/node"
+)
+
+const (
+	// EventConsoleLog defines the event type for the console.log call.
+	EventConsoleLog = "vm.console.log"
+
+	// EventConsoleWarn defines the event type for the console.debug call.
+	EventConsoleWarn = "vm.console.warn"
+
+	// EventConsoleDebug defines the event type for the console.debug call.
+	EventConsoleDebug = "vm.console.debug"
+
+	// EventConsoleError defines the event type for the console.error call.
+	EventConsoleError = "vm.console.error"
+)
+
+// Extension takes a giving extension function and writer and returns a standard otto.Function
+// callable by a otto vm.
+func Extension(w io.Writer, ext func(otto.FunctionCall, io.Writer) otto.Value) func(otto.FunctionCall) otto.Value {
+	return func(fn otto.FunctionCall) otto.Value {
+		return ext(fn, w)
+	}
+}
```

Comment:
I think it's ok, but maybe we could change naming? For me, it's more like an adapter. We want to change `func (fn otto.FunctionCall, w io.Writer) otto.Value` signature into `func (fn otto.FunctionCall) otto.Value`. So maybe `func HandlerAdapter(...)` would be more descriptive? Also, we can add an usage example in the comment.

Another solution would be to create a `Console` struct. Something like this:
```go
type Console struct {
  w io.Writer
}
```

Then, `Log()` would look like this:
```go
func (c *Console) Log(fn otto.FunctionCall) otto.Value {
  // SendSignal stuff
  handleConsole(c.w, "console.log: %s", fn.ArgumentList)
  return otto.UndefinedValue()
}
```

and it would be still possible to pass it like `c.Log` as in Go this method is bound, unlike in JS.